### PR TITLE
Slight Access Tweaks for Bridge Crew and Atmos Tech

### PR DESCRIPTION
**Pull request by furrycactus**
- Gives Bridge Crew access to the Fuel Bay in the Hangar so they can refuel the Canary. I was told this was apparently an oversight.
- Gives Atmos Tech access to the Leviathan.
- Added a single debugger to each Atmos Tech locker, so that they can fix malfunctioning vending machines without needing gloves.
- Gave Atmos Tech access to the little dilapidated office adjacent to the INDRA that has the tables and empty canisters inside. Can't think of a reason they shouldn't have access to what's basically an under construction area with nothing important in it.
- Put a crate with components for a gas heater and gas cooler in the Atmos Tech locker room.
- Moved the circuit kit in the Engineer locker room to the Engineering foyer.

### CODE REVIEW

**Comment by furrycactus** [comment]
!review

**Comment by FluffyGhoster** [comment]
"Gives Atmos Tech access to Tech Storage. [...] Additionally, it gives them access to a single pair of gloves in the event there are no Engineers present."

Atmos techs having access to gloves was rejected in #14325 and their gloves removed in #17651 



"Added a single debugger to each Atmos Tech locker, so that they can fix malfunctioning vending machines without needing gloves."

I do not believe this is in the purview of the atmos techs to do, atmos techs work with atmospheric (gas) systems, and I don't think they are supposed to fix vendors (besides powering them off)

**Comment by furrycactus** [comment]
They aren't intended to take the gloves willy-nilly, but sometimes it's necessary if there are no engineers present, particularly on lowpop. If there are no engineers present in those situations, then fixing vending machines or wiring an emitter to handle a blob are tasks that are not possible without gloves or access to the electrical closets.

At the very least, fixing vendors feels like more of a chore or busywork than anything else. I do not think Engineers are going to be missing out on a core piece of their gameplay loop if atmos techs have access to a debugger to slowly fix a vending machine from pelting objects in their absence. And in addition, giving them a debugger removes a need for them to be messing with electrical systems in the first place, as the debugger just uploads a software patch to the vendors automatically.

**Comment by FluffyGhoster** [comment]
> They aren't intended to take the gloves willy-nilly, but sometimes it's necessary if there are no engineers present, particularly on lowpop. If there are no engineers present in those situations, then fixing vending machines or wiring an emitter to handle a blob are tasks that are not possible without gloves or access to the electrical closets.

If people to do a job are not present, it follows that the job cannot be done, eg. if there's no BC and relevant command, the ship cannot be moved, if there's no doctors, surgeries cannot be done, if there's no pharmacist, medicines cannot be produced, if there's no miners, resources cannot be mined and so on; it's a game that focuses on inter-reliances and interlocks between different slots to enable the accomplishment of some tasks, if essential slots for said task are not filled, the task cannot be accomplished, it doesn't currently give a greenlight for other roles to take over and do them (though it could, but it would need a policy change), ultimately my point however is to highlight that the access to insulated gloves for atmos techs was rejected by the maintainers in said PRs, the maintainers can decide to allow this or reverse said decision at their discretion, which is why I have not applied a change request on this PR, it is however likewise important to highlight the aforementioned rejection


> At the very least, fixing vendors feels like more of a chore or busywork than anything else. I do not think Engineers are going to be missing out on a core piece of their gameplay loop if atmos techs have access to a debugger to slowly fix a vending machine from pelting objects in their absence. And in addition, giving them a debugger removes a need for them to be messing with electrical systems in the first place, as the debugger just uploads a software patch to the vendors automatically.

Possibly, I do not know what the engineer's gameplay loop is as I do not play engineering myself, I think however that knowing how to fix vendors is outside the knowledge domain of atmos techs (eg. a machinist would have more reason to know about it than a technician who works on gasses, I guess we can equate it to a today's plumber vs electrician), an admin is free to correct me if I am wrong on this assumption

**Comment by furrycactus** [comment]
> If people to do a job are not present, it follows that the job cannot be done, eg. if there's no BC and relevant command, the ship cannot be moved, if there's no doctors, surgeries cannot be done, if there's no pharmacist, medicines cannot be produced, if there's no miners, resources cannot be mined and so on; it's a game that focuses on inter-reliances and interlocks between different slots to enable the accomplishment of some tasks, if essential slots for said task are not filled, the task cannot be accomplished, it doesn't currently give a greenlight for other roles to take over and do them

I understand what you are saying, and I agree with you in the case of a fully missing department, but while each job has an intended thing that they are supposed to specialise in, it's not accurate to imply there's no areas where responsibilities cross over for things that are either minor or necessary. In the absence of Officers, other security personnel process and brig/fine people. In the absence of Surgeons, Physicians do a small selection surgeries associated with common injuries (fractures, lungs, shrapnel, AB).

In the absence of a Hangar Tech, Engineers can load the ship guns. In the absence of a Shaft Miner, Engineers can produce metals with the INDRA (which they have sole purview over, which is fine, and also good for the record). In the absence of a Janitor, Engineers can and do hack into Custodial to replace broken lights when asked. In the absence of an Atmos Tech, Engineers can set cold Thrust and pressurise the ship's Air Supply. As it stands, Engineers already can cover a lot of tasks in the absence of other jobs, and that's fine, I'm not arguing to remove that. Right now, all Atmos Techs do is set the air supply and set thrusters, and then it's a lot of sitting on their hands. Which is somewhat fine, as Engineering is quite a reactive department after the essentials are set up. But I don't think it breaks precedent to suggest that they be able to fix some vendors with a debugger and wire up an emitter to deal with a very round-impacting event without burning their hands off in the absence of an Engineer.

This certainly isn't me arguing for a wholesale return of gloves to Atmos because I acknowledge that the delination is important, and understand the reason behind the rejection for Atmos having gloves by default. I simply feel access to a single pair for emergency situations should not be outside the realm of acceptability. Providing access to the debuggers as well is specifically to remove the need for Atmos to want gloves for something so trivial, again further pushing a need for gloves towards emergency situations only, since to my understanding the debuggers are quite literally there to provide a way to fix vendors without touching wires, at the cost of speed and efficiency, because they take much longer on average to fix the problem. (I know they originally fixed malf-hacked APCs, but we don't have malf anymore).

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]
The only way for atmos techs to have any sort of access to insulated gloves is to make them an engineer alt title. This decision isn't going to change.

**Comment by furrycactus** [comment]
> The only way for atmos techs to have any sort of access to insulated gloves is to make them an engineer alt title. This decision isn't going to change.

Fair enough then. Is access to the debugger still acceptable? 

Additionally, would it be acceptable to move the gas cooler/heater boards to somewhere that atmos tech does have access?

**Comment by NonQueueingMatt** [comment]
> > The only way for atmos techs to have any sort of access to insulated gloves is to make them an engineer alt title. This decision isn't going to change.
> 
> Fair enough then. Is access to the debugger still acceptable?
> 
> Additionally, would it be acceptable to move the gas cooler/heater boards to somewhere that atmos tech does have access?

Both are fine

**Review by NonQueueingMatt — APPROVED** [review]

File: code/game/jobs/job/engineering.dm [hunk 1]
```diff
@@ -163,8 +163,8 @@
 		SPECIES_SKRELL_AXIORI = 60
 	)
 
-	access = list(ACCESS_EVA, ACCESS_ENGINE, ACCESS_ENGINE_EQUIP, ACCESS_TECH_STORAGE, ACCESS_MAINT_TUNNELS, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_CONSTRUCTION, ACCESS_ATMOSPHERICS, ACCESS_EXTERNAL_AIRLOCKS)
-	minimal_access = list(ACCESS_EVA, ACCESS_ENGINE, ACCESS_ATMOSPHERICS, ACCESS_MAINT_TUNNELS, ACCESS_EMERGENCY_STORAGE, ACCESS_CONSTRUCTION, ACCESS_EXTERNAL_AIRLOCKS)
+	access = list(ACCESS_EVA, ACCESS_ENGINE, ACCESS_ENGINE_EQUIP, ACCESS_TECH_STORAGE, ACCESS_MAINT_TUNNELS, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_CONSTRUCTION, ACCESS_ATMOSPHERICS, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_LEVIATHAN)
+	minimal_access = list(ACCESS_EVA, ACCESS_ENGINE, ACCESS_MAINT_TUNNELS, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_CONSTRUCTION, ACCESS_ATMOSPHERICS, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_LEVIATHAN)
```
Comment (DreamySkrell) [inline code/game/jobs/job/engineering.dm:167]: image
ACCESS_EXTERNAL_AIRLOCKS is repeated twice here

**Review by DreamySkrell — CHANGES_REQUESTED** [review]

**Review by DreamySkrell — APPROVED** [review]

**Pull request merged**